src/components/IdealImage/index.js

```diff
@@ -320,8 +323,10 @@ export default class IdealImage extends Component {
       ...this.state,
       size: pickedSrc.size,
     })
-    this.setState({pickedSrc, shouldAutoDownload, url})
-    if (shouldAutoDownload) this.load(false)
+    this.setState(
```

Comment:
It's important for this to be callback otherwise this.load could have stale state (I encountered this with url being undefined)